tools/yulPhaser/Population.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <tools/yulPhaser/Population.h>
+
+#include <tools/yulPhaser/Program.h>
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <numeric>
+
+using namespace std;
+using namespace solidity::langutil;
+
+namespace solidity::phaser
+{
+
+size_t Population::measureFitness(Chromosome const& _chromosome, CharStream& _sourceCode)
+{
+	auto program = Program::load(_sourceCode);
```

Comment:
Can the result of `Program::load` be stored across multiple chromosomes? Is that what you are currently working on with the ASTCopier?

Reply:
Yes, that's the idea. I'll have one "master copy" obtained from `load()` and then `measureFitness()` will copy it and only run `optimize()` on it.

Reply:
Ok!

Reply:
The PR is here: #8223.